code/game/jobs/job/job.dm

```diff
@@ -176,6 +177,14 @@
 		return max(0, minimal_player_age - C.player_age)
 	return 0
 
+/datum/job/proc/is_job_whitelisted(client/C)
+	if (C && config.use_job_whitelists && require_whitelist)
+		if (whitelist_lookup("AI", C.ckey) || !isnull(C.holder))
```

Comment:
Hardcoding the whitelist key for any job to be "AI" should be self-explanatorily bad. Could probably create a `whitelist_key` var for each whitelisted job and use that instead.